gridstatus/tests/vcr_utils.py

```diff
@@ -0,0 +1,77 @@
+import json
+import os
+import shutil
+from urllib.parse import parse_qs, urlparse
+
+import vcr
```

Comment:
Do we need to add this library to the dev dependencies in `pyproject.toml`?

Reply:
ah yeah we do, still learning poetry a bit, will definitely add.

Reply:
Looks like there is a conflict (though I definitely didn't run into any usage issues that I could tell), so I will add some dev dep logic and we'll see where we stand.

Makes me a little nervous that VCR hasn't upgraded to 2.0+ of urllib3. Might explore if there are other packages that do something similar.

Reply:
Seems like there is some motion, but not a ton. https://github.com/kevin1024/vcrpy/pull/777 And it is a much more active project than [betamax](https://github.com/kevin1024/vcrpy/pull/777), which is similar.

But ultimately it seems there is no resolution for existing `urllib3` and `vcrpy` dependencies. Very frustrating. 
![CleanShot 2024-10-18 at 14 14 52@2x](https://github.com/user-attachments/assets/97636830-165b-4fd9-8c6e-817fca273c13)

Reply:
Yeah, unfortunately with [no ability to specify conditional package installation](https://stackoverflow.com/questions/76102868/how-to-specify-2-diff%C3%A9rents-source-for-a-same-package-when-in-prod-or-in-dev-whe) in poetry and no resolution between urllib3 and vcr, I can't put vcr in the pyproject.toml and have it resolve.

Right now the options are
1. forgo vcr entirely (sad)
2. downgrade urllib3 to 1.25.0 (which might have other effects)
3. put the installation recommendation for vcr in the readme only, and open an issue on vcr to support 2.0+ urllib3. (path of least resistance, and again I didn't have any issues with urllib3==2.2.2 and vcrpy==6.0.2 when developing, so is likely the best option right now)
4. switch to a more flexible, modern package manager like `uv` [which allows conditional multi-source dependencies.](https://docs.astral.sh/uv/concepts/dependencies/#multiple-sources) (this would be my long term desire, but probably out of scope for this PR)